srcts/src/shiny/bind.ts

```diff
@@ -47,45 +48,54 @@ function valueChangeCallback(
  * immediately invoked function to keep the sets private and not clutter the
  * scope.
  */
-const bindingsRegistery = (() => {
+const bindingsRegistry = (() => {
+  type BindingTypes = "input" | "output";
+
   /**
    * Keyed by binding IDs to the array of each type of binding that ID is associated for in current app state.
    *
    * Ideally the
    * value would be a length 1 array but in some (invalid) cases there could be
    * multiple types for a single ID.
    */
-  type IdToBindingTypes = Map<string, Array<"input" | "output">>;
+  type IdToBindingTypes = Map<string, BindingTypes[]>;
 
   // Main store of bindings.
   const bindings: IdToBindingTypes = new Map();
 
   /**
-   * Checks if the bindings registery is valid. Currently this just checks for
+   * Checks if the bindings registry is valid. Currently this just checks for
    * duplicate IDs but in the future could be expanded to check more conditions
-   * @returns ShinyClientError if current ID bindings are invalid, otherwise null
+   * @returns ShinyClientError if current ID bindings are invalid, otherwise
+   * returns an ok status.
    */
   function checkValidity():
     | { status: "error"; error: ShinyClientError }
     | { status: "ok" } {
-    const duplicateIds: IdToBindingTypes = new Map();
+    const duplicateIds: Map<string, { input: number; output: number }> =
+      new Map();
+
+    const countBindingType = (
+      types: BindingTypes[],
+      bindingType: BindingTypes
+    ) => {
+      return types.filter((type) => type === bindingType).length;
+    };
```

Comment:
```suggestion
    type BindingCounts = { [T in BindingTypes]: number };
    const duplicateIds = new Map<string, BindingCounts>();

```

Reply:
Thanks, that is a nice way to have everything flow through the `BindingTypes` type!